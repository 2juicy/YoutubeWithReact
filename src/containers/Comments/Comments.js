import React from "react";
import CommentsHeader from "./CommentsHeader/CommentsHeader";
import Comment from "./Comment/Comment";
import AddComment from "./AddComment/AddComment";

function Comments({ amount }) {
  return (
    <div>
      <CommentsHeader amount={amount} />
      <AddComment />
      <Comment />
      <Comment />
      <Comment />
      <Comment />
    </div>
  );
}

export default React.memo(Comments);
