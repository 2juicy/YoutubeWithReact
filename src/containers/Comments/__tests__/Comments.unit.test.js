import React from "react";
import { shallow } from "enzyme";
import Comments from "../Comments";

describe("Comments", () => {
  test("renders without props", () => {
    const wrapper = shallow(<Comments />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with amount", () => {
    const wrapper = shallow(<Comments amount={12345} />);
    expect(wrapper).toMatchSnapshot();
  });
});
