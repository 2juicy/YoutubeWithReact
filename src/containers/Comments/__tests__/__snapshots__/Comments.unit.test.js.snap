// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders with amount 1`] = `
<div>
  <Comments
    amount={12345}
  />
  <AddComment />
  <Comment />
  <Comment />
  <Comment />
  <Comment />
</div>
`;

exports[`Comments renders without props 1`] = `
<div>
  <Comments />
  <AddComment />
  <Comment />
  <Comment />
  <Comment />
  <Comment />
</div>
`;
